app/validators/belongs_to_host_taxonomy_validator.rb

```diff
@@ -8,18 +8,14 @@ def validate_each(record, attribute, value)
     taxonomy = @options[:taxonomy]
     return unless record.host.present? && value.present?
 
-    host_taxonomy = record.host.public_send(taxonomy)
-    attribute_taxonomies = value.public_send(taxonomy.to_s.pluralize.to_sym)
+    host_taxonomy_ids = record.host.public_send(taxonomy)&.path_ids
+    attribute_taxonomy_ids = value.public_send("#{taxonomy}_ids")
 
-    return if host_taxonomy.nil? && attribute_taxonomies.empty?
+    return if host_taxonomy_ids.nil? && attribute_taxonomy_ids.empty?
```

Comment:
I think `path_ids` returns array so maybe `host_taxonomy_ids.empty?` ?

Reply:
That's true, but than, it should be always defined if the host has taxonomy, so we can have `nil` or not empty array - at least self id should always be in the array.